pennylane/tape/qscript.py

```diff
@@ -891,27 +898,36 @@ def _shape_new(self, device):
             >>> qs.shape(dev)
             ((4,), (), (4,))
         """
-        if device.shot_vector is not None and self.batch_size is not None:
+        if not qml.active_return():
+            return self._shape_legacy(device)
+
+        shots = (
+            Shots(device._raw_shot_sequence)
```

Comment:
Does something fail if you do Shots(device.shot_vector)? Shots should be able to handle the condensed representation.